aten/src/ATen/NamedTensorUtils.h

```diff
@@ -37,6 +38,7 @@ void propagate_names(Tensor& result, const Tensor& src);
 void propagate_names(TensorImpl* result, /*const */TensorImpl* src);
 
 void propagate_names_except(Tensor& result, const Tensor& src, IntArrayRef excluded_idxs);
+void propagate_names_for_reduction(Tensor& result, const Tensor& src, IntArrayRef excluded_idxs, bool keepdim);
```

Comment:
might be helpful to have a code comment explaining the different name propagation semantics and when/where each of these three are used